scripts/apps/dashboard/user-activity/components/UserActivityWidget.tsx

```diff
@@ -0,0 +1,411 @@
+import React from 'react';
+import {WidgetItemList} from 'apps/search/components';
+import {IArticle, IUser, IStage} from 'superdesk-api';
+import {AuthoringWorkspaceService} from 'apps/authoring/authoring/services';
+import ng from 'core/services/ng';
+import {gettext} from 'core/utils';
+import {SelectUser} from 'core/ui/components/SelectUser';
+
+interface IGroup {
+    id: string;
+    label: string;
+    repo?: string; // not used if self.query is a function
+    // pass the query or build multiple queries with the provided function
+    query: any | ((fetchFn: (repo: string, criteria: any) => Promise<any>) => Promise<any>);
+    collapsed?: boolean;
+}
+
+interface IState {
+    searchOpen: boolean;
+    userSearchField: string;
+    user?: IUser;
+    groups: Array<IGroup> | null;
+    groupsData: Array<{ id; itemIds; itemsById }> | null;
+    loading: boolean;
+}
+
+function genericFetch({search, api}, repo, filters) {
+    let query = search.query();
+
+    query.clear_filters();
+    query.size(1).filter(filters);
+
+    let criteria = query.getCriteria(true);
+
+    criteria.repo = repo;
+
+    return api.query('search', criteria);
+}
+
+function getQueryMarkedForUser(userId) {
+    return {
+        term: {
+            marked_for_user: userId,
```

Comment:
Up until now everything related to mark for user feature was contained in the extension and I'd like to keep it contained as much as possible.

Ideally, this widget would be an extension(I wouldn't convert it yet if it's a lot of work) and would expose a method like
```registerActivityType('Marked for user', () => {marked_for_user: user._id})```
which we could call when mark for user extension is initialized and it would register the filter.

The problem is `'Created by this user'` condition which is combining a bunch of other different conditions. In this case, could you try [exposing the queries from the extension](https://user-images.githubusercontent.com/2076953/72739943-fcca1980-3ba4-11ea-861b-a0210f93ee09.png)?


You'd need to modify [the IExtension type](https://user-images.githubusercontent.com/2076953/72740070-3c910100-3ba5-11ea-8cf0-85ba26de1380.png).

Reply:
It would be coupled to the extension, but it would be coupled explicitly and only to methods exposed by the extension. It would make discovering dependencies easy as well as removing the feature without leaving dead code since the compiler would raise errors.

It'd be a much better situation than it is with current features. If I asked you to remove "mark for desk" feature and make sure you don't leave dead code it would be quite a grep challenge with a "maybe" outcome. This is what I want to prevent by making dependencies explicit.